keps/sig-network/5311-relaxed-validation-for-service-names/README.md

```diff
@@ -0,0 +1,761 @@
+
+Since the relaxed check allows previously invalid values, care must be taken to support cluster downgrades safely. To accomplish this, the validation will distinguish between new resources and updates to existing resources:
+
+When the feature gate is disabled:
+
+- Creation of Services will use the previous RFC-1035 validation for `.metadata.name`
+- Creation of Ingress will use the previous RFC-1035 validation for `.spec.[]rules.http.[]paths.backend.service.name`
+- Updates of Services will use the new RFC-1123 validation for `.metadata.name`
+- Updates of Ingress will use the new RFC-1123 validation for `.spec.[]rules.http.[]paths.backend.service.name`
+
+When the feature gate is enabled:
+
+- Creation and update of Services will use new RFC-1123 relaxed validation for `.metadata.name`
+- Creation and update of of Ingress will use the new RFC-1123 validation for `.spec.[]rules.http.[]paths.backend.service.name`
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+The goal is to ensure that we don't accept enhancements with inadequate testing.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+[X] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+<!--
+Based on reviewers feedback describe what additional tests need to be added prior
+implementing this enhancement to ensure the enhancements have also solid foundations.
+-->
+
+##### Unit tests
+
+Tests of validation will be updated.
+
+<!--
+In principle every added code should have complete unit test coverage, so providing
+the exact set of tests will not bring additional value.
+However, if complete unit test coverage is not possible, explain the reason of it
+together with explanation why this is acceptable.
+-->
+
+<!--
+Additionally, for Alpha try to enumerate the core package you will be touching
+to implement this enhancement and provide the current unit coverage for those
+in the form of:
+- <package>: <date> - <current test coverage>
+The data can be easily read from:
+https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit
+
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+-->
+
+- `<package>`: `<date>` - `<test coverage>`
```

Comment:
fill this in for `pkg/apis/core/validation` and `pkg/apis/networking/validation`